packages/sdk/src/utils/is-valid-webhook-elba-signature.ts

```diff
@@ -0,0 +1,130 @@
+/**
+ * Validates the signature sent in the 'X-elba-Signature' header of a webhook request
+ * received from elba.
+ *
+ * This function computes the HMAC SHA256 signature of the given payload and secret,
+ * and then compares it in a timing-safe manner with the signature received in the webhook's
+ * 'X-elba-Signature' header. This is used to ensure that the webhook request is indeed from elba
+ * and has not been tampered with. The comparison is done using a timing-safe algorithm to prevent
+ * timing attacks.
+ *
+ * @param secret - The secret key generated at the time of creating the elba integration.
+ * @param payload - A JSON object with the payload for which the signature is being validated.
+ * @param elbaSignatureFromHeader - The signature to be validated against the computed signature.
+ * @returns A promise that resolves to true if the computed signature matches
+ *          the signature from the header; false otherwise.
+ *
+ * @example
+ * ```ts
+ * const secret = getSecurelyStoredSecret();
+ * const payload = { data: 'example' };
+ * const signatureFromHeader = 'received-signature-from-header';
+ * const isValid = await isValidWebhookElbaSignature(secret, payload, signatureFromHeader);
+ * console.log("Is the signature valid?, isValid ? "Yes" : "No");
+ * ```
+ */
+export async function isValidWebhookElbaSignature(
+  secret: string,
+  payload: Record<string, unknown>,
+  elbaSignatureFromHeader: string
+): Promise<boolean> {
+  const computedElbaSignature = await _createWebhookElbaSignature(secret, payload);
+  return _timingSafeEqual(computedElbaSignature, elbaSignatureFromHeader);
+}
+
+/**
+ * @internal
+ * Asynchronously generates a HMAC SHA256 signature for a given payload using a secret.
+ * Uses the SubtleCrypto API with the secret and a payload encoded to generate
+ * a HMAC SHA256 signature. The signature is returned as a hexadecimal string.
+ *
+ * @param secret - The secret key used for HMAC generation.
+ * @param payload - A JSON object with the payload for which the signature is being generated.
+ * @returns A promise that resolves to a hexadecimal string representation of the signature.
+ *
+ * @example
+ * ```ts
+ * const secret = getSecurelyStoredSecret();
+ * const payload = { data: 'example' };
+ * const signature = await createHmacSha256Signature(secret, payload)
+ * console.log(signature);
+ * ```
+ */
+export async function _createWebhookElbaSignature(
```

Comment:
```suggestion
export async function createWebhookElbaSignature(
```

![image](https://github.com/elba-security/elba-security/assets/117630727/fff90787-e97b-4fe5-8eb0-604b77288f3a)